### PR TITLE
Ftrack Id Module: adding parameter to callback in getId return object

### PR DESCRIPTION
**Pull request by ftxmoJason**
## Type of change
- [X] Other

## Description of change
1. Adding a `cb` parameter to the callback property in the object returned by getId() so that more pbjs get ID methods can be used
2. Tweaking the eids.js so that a value is returned. We tried to return our IDs as an object but had to JSON.stringify() it.

### CODE REVIEW

**Comment by ftxmoJason** [comment]
To build on @TheMediaGrid 's comment. Our IDs are formatted like so:
```
{
    "HHID": ["<USERS HH ID>"],
    "DeviceID": ["<USERS DEVICE ID>"],
    "SingleDeviceID": ["USERS SINGLE DEVICE ID"]
}
```
And we can't return an object from eids.js `getValue`, so we `JSON.stringify()` the object and return it. Is that the best approach?

**Comment by ftxmoJason** [comment]
@ftxmojared Just trying to ping you in this thread for awareness.

**Comment by patmmccann** [comment]
have you studied this https://github.com/prebid/Prebid.js/issues/8429

**Comment by ftxmoJason** [comment]
Thanks @patmmccann for the suggestion.
I tweaked the `createEidsArray` method and added a condition for 'ftrack', similar to how 'pubProvidedId' did it.

File: modules/userId/eids.js [hunk 1]
```diff
@@ -367,6 +368,18 @@ export function createEidsArray(bidRequestUserId) {
...
+        let eid = {
+          'atype': 1,
+          'id': (bidRequestUserId[subModuleKey]['DeviceID'] || []).join('|'),
+          'ext': {}
+        }
+        for (let id in bidRequestUserId[subModuleKey]) {
+          eid.ext[id] = (bidRequestUserId[subModuleKey][id] || []).join('|');
+        }
+
+        eids.push(eid);
```
Comment (smenzer) [inline modules/userId/eids.js:382]: this block needs tests cases, please

**Review by smenzer — CHANGES_REQUESTED** [review]
i did an initial pass through and first thing we need are test cases for the new functionality. also, are there any changes to publisher implementation of the config or does this only impact those reading the EIDs array?

File: modules/userId/eids.js (hunk 1 above)
Comment (TheMediaGrid) [reply modules/userId/eids.js:382]: no changes to the publisher config implementation, just the eids object interaction is changed
I will also let @ftxmoJason comment on the test cases

**Comment by TheMediaGrid** [comment]
@smenzer can you please take a look at the updates here? we'd really like this PR to make it to the next release if possible

**Review by smenzer — CHANGES_REQUESTED** [review]

File: test/spec/modules/ftrackIdSystem_spec.js [hunk 2]
```diff
@@ -298,4 +307,88 @@ describe('FTRACK ID System', () => {
...
+        userSync: {
+          auctionDelay: 10,
+          userIds: [{
+            name: 'ftrack', value: ids,
+          }]
+        }
+      });
+
+      getGlobal().getUserIdsAsync().then((ids) => {
+        expect(getGlobal().getUserIdsAsEids()).to.deep.equal(createEidsArray(ids));
```
[the remark this answers is not in the document]
Comment (smenzer) [reply test/spec/modules/ftrackIdSystem_spec.js]: thanks for the updates

**Review by smenzer — APPROVED** [review]
LGTM

This probably warrants an update to the prebid docs repository, but I'll go ahead and approve this now since this is good to go.

**Pull request merged**